app/src/server/tasks/generateTestSetEntry.task.ts

```diff
@@ -11,20 +12,27 @@ import {
   calculateFineTuneUsageCost,
   isComparisonModel,
 } from "~/utils/baseModels";
-import { calculateEntryScore } from "../utils/calculateEntryScore";
+import {
+  saveFieldComparisonScore,
+  calculateFieldComparisonScore,
+} from "../utils/calculateFieldComparisonScore";
 import { getOpenaiCompletion } from "../utils/openai";
 import defineTask from "./defineTask";
+import { queueHeadToHeadEvalJobsForTestingEntry } from "./evaluateTestSetEntries.task";
 
-export type EvaluateTestSetEntryJob = {
+export type GenerateTestSetEntryJob = {
   modelId: string;
   datasetEntryId: string;
   skipCache?: boolean;
 };
 
-export const evaluateTestSetEntry = defineTask<EvaluateTestSetEntryJob>({
-  id: "evaluateTestSetEntry",
+export const generateTestSetEntry = defineTask<GenerateTestSetEntryJob>({
+  id: "generateTestSetEntry",
   handler: async (task) => {
-    const { modelId, datasetEntryId, skipCache } = task;
+    const { modelId, datasetEntryId } = task;
+
+    // TODO: fix evals with caching
+    const skipCache = true;
```

Comment:
Under what circumstances will this cause a re-run of all outputs/evals?

Reply:
The purpose of this task is to query and save output created by a comparison or fine-tuned model based on input from the dataset entry. The cache is meant to be used in two ways:

1. Prevent this task from running on the same combination of dataset entry and model id as it has in the past
2. Save this task the work of re-retrieving output when presented with identical input used in a previous retrieval run. Instead of retrieving the output a second time from the same model with the same input, we copy over the output that was retrieved the first time we ran the task with this input and modelId.

In scenario 1, skipping the cache could indeed cause unnecessary and duplicate evals to be run. So if we call generateTestSetEntry on an identical dataset entry and model id many times, then we'll duplicate the evals many times as well. This would be very inefficient and costly. However, I don't believe our code is designed to call generateTestSetEntry multiple times with the same input.

In scenario 2, skipping the cache will result in an extra inference request to calculate the output for this model id and dataset entry combination, but will not result in extra eval calls. Since our evals are comparative (and since the explanation as currently designed requires the model's true name to be available while constructing it) we need to re-run these evals every time we generate new output, whether we generate it from a request to an LLM or from cached output.

Reply:
In any case, I fixed our evals to work with caching, so this should be an non-issue.